# Data Science, Bioinformatics, and Science
Welcome to my portfolio summarizing my scientific and Data Science related projects and skills. This repository highlights my expertise in **programming**, **data analysis**, **machine learning**, and **data visualization**.

---

## üìë Table of Contents
1. [Tech Stack](#-Tech-Stack)
2. [Machine Learning projects](#-Machine-Learning-projects)
3. [Bioinformatics projects](#-Bioinformatics-projects)
4. [Scientific publications](#-Scientific-publications)
5. [Scientific conferences](#-Scientific-conferences)
6. [Certificates](#-Certificates)

---

## üöÄ Machine Learning Projects

### 1. [Transformers towards detection of transposons](https://github.com/DMH-biodatasci/Transformers-towards-detection-of-Transposons)
**Objective:** Predict composite transposable elements in bacterial genomes.  
- **Tech Stack:** Python, Transformers, Huggingface, Named entity recognition, Masked Language Model, Pandas Numpy, Pickle, Matplotlib  
- **Highlights:**
  - **Challenging Detection of Composite Transposable Elements:** Boundaries of composite transposable elements are difficult to determine due to missing terminal features, diversity in genetic composition and size, and association with both essential and non-essential genes, which affects the gene frequency landscape.
  - **Innovative Approach Using Transformers:** The study employs Transformer-based models to tokenize protein families, implement a masked language model to understand bacterial genome rules, perform named entity recognition for transposable elements, and detect boundaries for validation.
  - **Advanced Computational Resources:** Training was conducted using Google Colab GPUs and NEC HPC-System (CAU) with 4 x NVIDIA Tesla V100 GPUs, leveraging multi-GPU training processes for scalability and performance.

### 2. [LSTM towards detection of transposons](https://github.com/DMH-biodatasci/LSTM-towards-detection-of-transposons)
**Objective:** Predict composite transposable elements in bacterial genomes.  
- **Tech Stack:** Python, TensorFlow, LSTM, Pandas, Seaborn, sklearn, Numpy, Matplotlib  
- **Highlights:**
  - **Complex Nature of Composite Transposable Elements:** These elements, characterized by flanking inverted repeats and transposable elements, move as single units within genomes, but their detection is challenging due to missing terminal features, diversity in composition and size, and associations with varying gene frequencies.
  - **Innovative Detection Using LSTM Models:** A supervised LSTM-based neural network model was developed to predict regions of composite transposable elements by analyzing gene frequencies, protein family clusters, and positional data.
  - **High-Performance Computational Resources:** The project utilized the NEC HPC-System (University of Kiel) with 4 x NVIDIA Tesla V100 GPUs for model training, with detailed resources and results provided in accompanying code and documentation.

### 3. [CNN towards Mushroom classification](https://github.com/DMH-biodatasci/CNN-towards-Mushroom-classification)
**Objective:** Classification of mushroom images to distinguish between edible and non-edible mushrooms.  
- **Tech Stack:** Python, CNN, Kaggle, Colab, TensorBoard, ImageDataGenerator, pickle, Matplotlib  
- **Highlights:**
  - **Mushroom Classification with CNNs:** Developed a convolutional neural network to classify mushroom genera, focusing on beginner-friendly groups (Russula and Boletales) to aid in identifying edible mushrooms.

---

## üõ†Ô∏è Tech Stack
### **Programming Languages:**
- **Python**, **R**, **Bash**

### **Machine Learning Frameworks:**
- **TensorFlow**, **Keras**, **PyTorch**, **Hugging Face Transformers**

### **Data Analysis:**
- **Pandas**, **NumPy**, **Scikit-learn**

### **Data Visualization:**
- **Matplotlib**, **Seaborn**, **Plotly**, **TensorBoard**

### **High-Performance Computing:**
- **Google Colab**, **NEC HPC System (University of Kiel)**, **NVIDIA Tesla GPUs**

### **Bioinformatics Tools:**
- **Biopython**, **BLAST**, **MMSeqs**, **Genome Assembly Tools (Prokka, SPAdes)**

---

## üì∞ Scientific Publications
- **Hanke DM, Dagan T.** ‚Äì In Preparation.
  SegMantX, a novel tool for detection of DNA duplications, uncovers a major role of transposable elements in plasmid gene duplication
- **Hanke DM, Wang Y, Dagan T.(https://academic.oup.com/nar/article/52/12/7049/7684600#)** ‚Äì Nucleic Acids Research (2024)  
  Large-scale bioinformatics analysis that uncovers a higher pseudogene density in plasmid compared to chromosomes and its biological implications (e.g., plasmid domestication).
- **Nies F, Wein T, Hanke DM, Springstein BL, Alcorta J, Taubenheim C, Dagan T.(https://enviromicro-journals.onlinelibrary.wiley.com/doi/10.1111/1758-2229.13203)** ‚Äì Environmental Microbiology (2023)  
  

---

## üéì Certificates
- **University Didactics**
  
- **Machine Learning Degree by opencampus (2023)**

- **Machine Learning with TensorFlow by opencampus (2021/2022)**

- **Transformers for Natural Language Processing and Beyond by opencampus (2022)**

- **ML project manager @ Coding.Waterkant Hackathon (2022)**

- **Deep Learning by opencampus (2022/2023)**

- **Coursera: Neural Networks & Deep Learning**

- **Coursera: Deeplearning.AI TensorFlow Developer**
  - **Introduction to TensorFlow for Artifical Intelligence, Machine Learning, and Deep Learning**
  - **Convolutional Neural Networks in TensorFlow**
  - **Natural Language Processing in TensorFlow**
  - **Sequences, Time Series and Prediction**

- **Projectmanagement by opencampus**

- **Webdevelopement by opencampus (HTML, CSS, Javascript & Bootstrap)**  


---

## üåê Explore More
Check out my full portfolio on my [GitHub Pages website](https://dmh-biodatasci.github.io/Portfolio-Dustin-Martin-Hanke/).

Feel free to reach out if you'd like to learn more about me or my projects. üòä
